cmd/agent/subcommands/flare/command.go

```diff
@@ -91,6 +93,13 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 				)),
 				flare.Module,
 				diagnosesendermanagerimpl.Module,
+				// We need inventoryagent to fill the status page generated by the flare.
+				inventoryagent.Module,
+				// inventoryagent require a serializer. Since we're not actually sending the payload to
+				// the backend a nil will work.
+				fx.Provide(func() serializer.MetricSerializer {
```

Comment:
If another component needs a real serializer the code will panic.
Ideally, can `inventoryagent` be split into components so that the component you need for flare doesn't use the serializer? 
Otherwise, can you return `Optional[serializer.MetricSerializer]` and use that inside inventory (As serializer is optional) so that another component that would need a MetricSerializer will not crash? What do you think?

Reply:
As decided during our daily, I'll afdress this in another PR :+1: